publish.js

```diff
@@ -51,33 +48,38 @@ function publish (manifest, tarball, opts) {
       const metadata = buildMetadata(
         spec, auth, reg, pubManifest, tardata, opts
       )
-      return npmFetch(spec.escapedName, opts.concat({
+
+      return npmFetch(spec.escapedName, {
+        ...opts,
         method: 'PUT',
         body: metadata,
         ignoreBody: true
-      })).catch(err => {
+      }).catch(err => {
         if (err.code !== 'E409') { throw err }
-        return npmFetch.json(spec.escapedName, opts.concat({
+        return npmFetch.json(spec.escapedName, {
+          ...opts,
           query: { write: true }
-        })).then(
+        }).then(
           current => patchMetadata(current, metadata, opts)
         ).then(newMetadata => {
-          return npmFetch(spec.escapedName, opts.concat({
+          return npmFetch(spec.escapedName, {
+            ...opts,
             method: 'PUT',
             body: newMetadata,
             ignoreBody: true
-          }))
+          })
         })
       })
     })
   }).then(() => true)
 }
 
 function patchedManifest (spec, auth, base, opts) {
+  const { npmVersion } = opts
   const manifest = cloneDeep(base)
   manifest._nodeVersion = process.versions.node
```

Comment:
Should this be `opts.nodeVersion`?  We allow overriding the effective node version for installation, but maybe we ought not to for publishing?